pkg/kubelet/rktshim/cli.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rktshim
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	utilexec "k8s.io/kubernetes/pkg/util/exec"
+)
+
+var (
+	errFlagTagNotFound           = errors.New("arg: given field doesn't have a `flag` tag")
+	errStructFieldNotInitialized = errors.New("arg: given field is unitialized")
+)
+
+// TODO(tmrts): refactor these into an util pkg
+// Uses reflection to retrieve the `flag` tag of a field.
+// The value of the `flag` field with the value of the field is
+// used to construct a POSIX long flag argument string.
+func getLongFlagFormOfField(fieldValue reflect.Value, fieldType reflect.StructField) (string, error) {
+	flagTag := fieldType.Tag.Get("flag")
+	if flagTag == "" {
+		return "", errFlagTagNotFound
+	}
+
+	if fieldValue.IsValid() {
+		return "", errStructFieldNotInitialized
+	}
+
+	switch fieldValue.Kind() {
+	case reflect.Bool:
```

Comment:
I think the Bool type can also fallthrough the Int type and string type. Question here: Can we use `fmt.Sprintf("--%v=%v", flagTag, fieldValue)` for all these?